packages/vite/src/vite.ts

```diff
@@ -27,6 +28,23 @@ export async function bundle (nuxt: Nuxt) {
   const useAsyncEntry = nuxt.options.experimental.asyncEntry ||
     (nuxt.options.vite.devBundler === 'vite-node' && nuxt.options.dev)
   const entry = await resolvePath(resolve(nuxt.options.appDir, useAsyncEntry ? 'entry.async' : 'entry'))
+
+  let allowDirs = [
+    nuxt.options.appDir,
+    nuxt.options.workspaceDir,
+    ...nuxt.options._layers.map(l => l.config.rootDir),
+    ...nuxt.apps.default?.components.map(c => dirname(c.filePath)),
+    ...nuxt.apps.default?.plugins.map(p => dirname(p.src)),
+    ...nuxt.apps.default?.middleware.map(m => dirname(m.path)),
```

Comment:
To avoid future refactor, shall we iterate over all possible `apps` keys?

Reply:
We might try multi-app via an experimental module as well. If core (or any modules might start depending on `nuxt.apps` can be ready for it, it could be a bonus  :)

Reply:
sure, makes sense - but I think it's worth noting (for the benefit of any modules) that we definitely do not support multi-app yet. we still need a team discussion first.

Reply:
For sure :) It is only placeholder for the future.